Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -151,9 +151,10 @@ void TextureCache::Cleanup(int _frameCount)
 		{
 			iter->second->frameCount = _frameCount;
 		}
-		if (_frameCount > TEXTURE_KILL_THRESHOLD + iter->second->frameCount &&
-		    // EFB copies living on the host GPU are unrecoverable and thus shouldn't be deleted
-		    !iter->second->IsEfbCopy())
+		// Only remove EFB copies when they wouldn't be used anymore(changed hash), because EFB copies living on the
+		// host GPU are unrecoverable. Perform this check only every TEXTURE_KILL_THRESHOLD for performance reasons
+		if ((_frameCount - iter->second->frameCount) % TEXTURE_KILL_THRESHOLD == 1 && (!iter->second->IsEfbCopy() ||
```

Comment:
This way the frameCount for the cache entry would not contain the info when it was last used anymore. Like some other parts of the code check which entry can be removed, by choosing the oldest one. With this, the most likely oldest one, would become a fairly new entry. Also, who knows what this info could be used in the future.

But the logic is a bit more complicated than i would like it, yeah.